packages/metadata-editor/src/MetadataEditor.tsx

```diff
@@ -192,74 +183,124 @@ export class MetadataEditor extends ReactWidget {
     }
   }
 
-  onAfterShow(): void {
-    if (!this.editor) {
+  onUpdateRequest(msg: Message): void {
+    super.onUpdateRequest(msg);
+    if (!this.editor && this.metadata['code']) {
       const getMimeTypeByLanguage = this.editorServices.mimeTypeService
         .getMimeTypeByLanguage;
       this.editor = this.editorServices.factoryService.newInlineEditor({
-        host: document.getElementById('Code:' + this.name),
+        host: document.getElementById('code:' + this.id),
         model: new CodeEditor.Model({
-          value: this.getFormItem('Code').value,
+          value: this.metadata['code'].join('\n'),
           mimeType: getMimeTypeByLanguage({
-            name: this.getFormItem('Language').value.choice,
-            codemirror_mode: this.getFormItem('Language').value.choice
+            name: this.metadata['language'],
```

Comment:
This makes the assumption that `language` and `code` exist for all code types, should there be a check for these values somewhere?